app/api/items/add_item.go

```diff
@@ -57,11 +56,16 @@ type item struct {
 	GroupCodeEnter  *bool      `json:"group_code_enter" sql:"column:groupCodeEnter"`
 }
 
+type itemWithRequiredType struct {
```

Comment:
No, sorry. What you're suggesting will not work since we ignore non-exported fields and fields without the 'json' tag. Note that 'Type' is not twice in the struct (I removed it from the 'item' struct)